### ISSUE TITLE
The extension gets stuck while setting the session key transaction

### ISSUE THREAD

**Issue by mohakagr** [polkadot-js/extension#148]
OS: Mac OS Mojave 10.14.6
Brave Browser: Version 0.68.132 Chromium: 76.0.3809.132 (Official Build) (64-bit)
Chrome Browser: Version 76.0.3809.132 (Official Build) (64-bit)
Polkadot{.js} extension: 0.10.1

Whenever I try to "Set Session Key" on [https://polkadot.js.org/apps/#/staking/actions](https://polkadot.js.org/apps/#/staking/actions), the polkadot.js extension gets hanged/stuck. It opens a blank window and clicking on the extension also opens a blank window. Tried this multiple times on brave browser as well as chrome, same problem every time.

<img width="709" alt="Screenshot 2019-09-13 at 7 11 02 PM" src="https://user-images.githubusercontent.com/9555313/64871839-afbcc080-d663-11e9-8767-11ab94dff5ae.png">

> Console output after pression on "Set Session key"

<img width="556" alt="polka" src="https://user-images.githubusercontent.com/9555313/64871858-b9debf00-d663-11e9-8ac2-136c80656055.png">

**Comment by jacogr** [comment]
If you just open the extension via the menu popup, does it work? 

Assuming it has been updated to the latest version (which is in store) and it open via the icon, does options -> open in new window work?

Under chrome://extensions is there only 1 version?

**Comment by jacogr** [comment]
Assuming the popup works - can accept there as well (or in the new tab if opened and it works). 

Really stumped on this one.

**Comment by jacogr** [comment]
One additional comment of things to try - when you exit Chrome completely and the re-open does the issue persist?

**Comment by mohakagr** [comment]
> If you just open the extension via the menu popup, does it work?
> 
> Assuming it has been updated to the latest version (which is in store) and it open via the icon, does options -> open in new window work?
> 
> Under chrome://extensions is there only 1 version?

Yes, only one version: 0.10.1

**Comment by mohakagr** [comment]
> Assuming the popup via icon works - can accept there as well (or in the new tab if opened and it works).
> 
> Really stumped on this one.
> 
> Last question - when was the last time it did work?

It gets stuck. A blank window opens up after that and so the option tab doesn't even appear.
It worked today itself when bonding tx was done.

**Comment by mohakagr** [comment]
> One additional comment of things to try - when you exit Chrome completely and the re-open does the issue persist?

Yes, it only opens a black window if I click on the extension. To make it work again, I have to restart the browser, disable the extension and then reenable it. Also, all this time the extension icon show "1", like there is a pending transaction I believe.

**Comment by jacogr** [comment]
So it got itself into a non working loop - I will do some googling around that error and what can possibly be the cause. 

(Would be even weirder if after a browser restart it is still completely dead)

**Comment by mohakagr** [comment]
Restarting the browser makes the plugin workable again but the moment I initiate the set session key, it becomes blank again.
This time I opened the extension in a new window before initiating the transaction. As soon as I initiated the tx, it became blank and threw the following error in console: 

<img width="555" alt="Screenshot 2019-09-14 at 10 57 49 AM" src="https://user-images.githubusercontent.com/9555313/64903884-e5eb5600-d6de-11e9-9c3e-af946d3fa96d.png">

**Comment by jacogr** [comment]
Ok. Now I know what causes this. It doesn’t understand that the session keys have 4 entries. 

So the good news is - this can be fixed, and needs to be. The bad news is that it will need an update, so will be a couple of days. 

Will get the fix in this morning, but will only publish on Monday, then we have to wait a couple of days for approval. 

I really appreciate your help on this. Sorry about the inconvenience, this is horrible.

**Comment by mohakagr** [comment]
No problem! Thanks for your immediate action. I am wondering how did other people set their session key?

Also, as it will take a couple of days, what are the other ways by which I can set my session key today?

**Comment by jacogr** [comment]
So we actually do handle this case - or rather we are supposed to cater for exactly this. In the latest version we did update the metadata for all the chains, so I believe something must have gone wrong with this update here. We made changes to handling the decoding. Really frustrated by this fact since things should just work, but sadly, here we are.

Tl;DR I believe it broke in the last update - and only setkeys are effected.

So there are a couple of short-term avenues for today -

- temporarily disable the extension, add the seed to the apps UI, make the transaction, remove the account from the apps UI, re-enable the extension (Or even do this in incognito, where the extension is not available and the storage is cleared straight after exiting)

- Will have a fix in in the next 60 minutes (give or take), then you could actually run build and add the extension in dev mode. (I would only recommend this if you are used to messing around with repos - the process is straight forward, but I certainly don't expect normal non-dev users to actually do this). The only trick here is that you once again need to enable/disable since you will have 2 extensions.
